packages/nx/src/lock-file/yarn-parser.ts

```diff
@@ -227,7 +227,7 @@ function groupDependencies(
   const resolutionMap = new Map<string, YarnDependency>();
   const snapshotMap = new Map<YarnDependency, Set<string>>();
   Object.entries(dependencies).forEach(([key, snapshot]) => {
-    const resolutionKey = `${snapshot.resolution}${snapshot.integrity}`;
```

Comment:
This was a typo - grouping is only created for Yarn Classic, which has `resolved`, not for Yarn Berry which has a `resolution` field